commit 19de297a4344b21a72772f49bc4a3c9e26e3810f (refs/changes/20/4420/4)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-06-22T23:38:04+00:00 (1 year, 4 months ago)
    
    MANTA-3785 waferlock allowing destroyed VMs
    MANTA-3786 waferlock doesn't deal with old VMs with no "ips" property on NICs
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/sapi.js b/lib/sapi.js
index c9e4ef2..b3e254a 100644
--- a/lib/sapi.js
+++ b/lib/sapi.js
@@ -249,10 +249,21 @@ SapiPoller.prototype.state_runq_do = function (S) {
 		mod_assert.arrayOfObject(objs);
 
 		objs.forEach(function (vm) {
-			var ips = [];
+			if (vm.state === 'destroyed' || vm.destroyed)
+				return;
+			if (vm.state === 'failed')
+				return;
+			if (!vm.nics || !Array.isArray(vm.nics))
+				return;
 
+			var ips = [];
 			vm.nics.forEach(function (nic) {
-				nic.ips.forEach(function (ip) {
+				if (typeof (nic) !== 'object')
+					return;
+				var ipa = nic.ips;
+				if (ipa === undefined)
+					ipa = [nic.ip];
+				ipa.forEach(function (ip) {
 					var parts = ip.split('/');
 					ips.push(parts[0]);
 				});
